src/inc/conpty-universal.h

```diff
@@ -30,18 +30,161 @@
 
 const unsigned int PTY_SIGNAL_RESIZE_WINDOW = 8u;
 
-HRESULT CreateConPty(const std::wstring& cmdline,       // _In_
-                     const unsigned short w,            // _In_
-                     const unsigned short h,            // _In_
-                     HANDLE* const hInput,              // _Out_
-                     HANDLE* const hOutput,             // _Out_
-                     HANDLE* const hSignal,             // _Out_
-                     PROCESS_INFORMATION* const piPty); // _Out_
+// A case-insensitive wide-character map is used to store environment variables
+// due to documented requirements:
+//
+//      "All strings in the environment block must be sorted alphabetically by name.
+//      The sort is case-insensitive, Unicode order, without regard to locale.
+//      Because the equal sign is a separator, it must not be used in the name of
+//      an environment variable."
+//      https://docs.microsoft.com/en-us/windows/desktop/ProcThread/changing-environment-variables
+//
+struct WStringCaseInsensitiveCompare
+{
+    bool operator()(const std::wstring& lhs, const std::wstring& rhs) const
+    {
+        return (::_wcsicmp(lhs.c_str(), rhs.c_str()) < 0);

```

Comment:
I don't believe that we can use `_wcsicmp()` if we want to eventually be a UWP application. I'm not sure how important that is for now, @DHowett-MSFT opinions?

https://docs.microsoft.com/en-us/cpp/cppcx/crt-functions-not-supported-in-universal-windows-platform-apps?view=vs-2019

Reply:
I believe you're thinking of _mbsicmp and _mbsicmp_l.

_wcsicmp is part of a compliant ISO C++ implementation, and is explicitly documented as being [available to UWP applications](https://docs.microsoft.com/en-us/uwp/win32-and-com/win32-extension-apis
):

```
_wcsicmp | Introduced into api-ms-win-crt-string-l1-1-0.dll in 10.0.10240.
_wcsicmp_l | Introduced into api-ms-win-crt-string-l1-1-0.dll in 10.0.10240.
```

Technically, I think you're supposed to check for the API set being present, but I can't imagine the crt-string not being present.

Thoughts?

Reply:
I must've read it wrong, ignore me.